javascript-carousel/main.js

```diff
@@ -0,0 +1,55 @@
+let currentImage = 0;
+const $images = document.querySelectorAll(".image");
+const $dots = document.querySelectorAll(".dot");
+const $next = document.querySelector(".next");
+const $prev = document.querySelector(".prev");
+
+const renderImage = function (imageNo) {
+    $images.forEach(function(image) {
+        image.style.display = "none";
+    });
+
+    $dots.forEach(function(dot) {
+        dot.classList.remove("active");
+    });
+
+    $images[imageNo].style.display = "block";
+    $dots[imageNo].classList.add("active");
+}
+
+const next = function () {
+    currentImage += 1;
+    if (currentImage >= $images.length) {
+        currentImage = 0;
+    }
+    renderImage(currentImage);
+}
+
+const prev = function () {
+    currentImage -= 1;
+    if (currentImage < 0) {
+        currentImage = $images.length - 1;
+    }
+    renderImage(currentImage);
+}
+
+const dotImage = function (dot, index) {
+    dot.addEventListener('click', function () {
+        currentImage = index;
+        renderImage(index);
+    })
+}
+
+document.addEventListener('DOMContentLoaded', renderImage(currentImage));
+$next.addEventListener('click', next);
+$prev.addEventListener('click', prev);
+
+$dots.forEach(dotImage) 
+
+setInterval(function() { 
+    currentImage += 1;
+    if (currentImage >= $images.length) {
+        currentImage = 0;
+    }
+    renderImage(currentImage);
+ }, 3000);
```

Comment:
The auto-step interval should reset any time the user manually changes the active slide:

---

![image](https://user-images.githubusercontent.com/16143771/123737180-5e40ee00-d857-11eb-85a7-4324e26f39a2.png)

Reply:
Made the changes to reset the interval when user clicks on the dots or prev/next